### ISSUE TITLE
Report not synced entries - time entries with errors

### ISSUE THREAD

**Issue by kepi** [ozguryazilimas/redmine_toggl#4]
Would it be possible to add new report with sync errors? It would help users to notice that their time were not synced because of some error. It would be great to have also reason (error text) but at least knowing there was error is enough.

I'm not sure if there are regular use cases for this. We hit it because we have couple modifications prohibiting users from adding time entries. I.e. to issues which are closed.

Here is the code so you can see how it is rejected.

```ruby
module TimeEntryClosedIssue
  def self.included(base) # :nodoc:
    base.send(:include, InstanceMethods)

    base.class_eval do
      unloadable
      alias_method_chain :validate_time_entry, :closed_issue
    end
  end

  module InstanceMethods
    def validate_time_entry_with_closed_issue()
      validate_time_entry_without_closed_issue

      status = issue.status
      closed = status.is_closed? && [5,6].include?(status.id)

      # if issue is really closed (5,6 states) and user is not admin (no supervise time entries) user can't edit
      errors.add :issue, :issue_already_closed if closed && !User.current.global_permission_to?(:supervise_time_entries)
    end
  end
end
```

Would you consider to add support for handling errors like this?

Thanks

**Comment by zaburt** [comment]
Thank you for reaching us. What is the intended behavior here, do you have anything specific in mind ? Thinking out loud, when we say report, if we are talking about a screen showing previous errors, that would require storing the errors in the database. I would prefer not to store errors in the database (or in a file etc.) especially since an error may become obsolete once conditions change (e.g. a closed story can be opened).

Perhaps we can inform when the rake task is run and error happens. Better output for the rake task when an issue happens ? A specific keyword for logging ? Or maybe send email to the entry owner with the error message in the body ? Note that this will do the "informing" everytime the rake task is run, so if I create a bad entry, and I run rake task every 10 minutes, then I will be notified every 10 minute but I think that is acceptable.

**Comment by kepi** [comment]
Informing during rake task run should be totally OK. It would be great to have:

- new line for every error
- keyword to distinguish between rest of output
- entry ID
- entry time
- user
- time value

With this it would be pretty easy to just parse the logs and inform our users from cron.

I think that new mail for every rake run is acceptable but may get out of hand (weekends, holidays etc). IMHO error output is better and more universal.

Thanks for super fast response

**Comment by zaburt** [comment]
Thank you! Will make the changes and inform here when it is ready for testing.

**Comment by zaburt** [comment]
Made some changes on `4_more_details_for_errors_master` branch, can you give it a try ? The output is like this, note that when you run rake tasks to sync we are putting errors to stderr

    ERROR FOR USER: user_login MESSAGE: exception_message TOGGL_ENTRY: toggl_entry_attributes_we_received

**Comment by zaburt** [comment]
the branch is merged to master, will make a 2.6.0 release soon
